cmd/kubeadm/app/util/config/common.go

```diff
@@ -193,3 +193,36 @@ func MigrateOldConfig(oldConfig []byte) ([]byte, error) {
 
 	return bytes.Join(newConfig, []byte(constants.YAMLDocumentSeparator)), nil
 }
+
+// appendComponentFeatureGate will append and enable the specified featureName.
+func appendComponentFeatureGate(extraArgs map[string]string, featureName string) map[string]string {
+	if !isComponentFeatureGatePresent(extraArgs, featureName) {
+		if extraArgs == nil {
+			extraArgs = make(map[string]string)
+		}
+		if _, present := extraArgs["feature-gates"]; present && len(extraArgs["feature-gates"]) > 0 {
+			extraArgs["feature-gates"] = strings.TrimSuffix(extraArgs["feature-gates"], ",")
```

Comment:
interesting, so this handles a stray `,` at the end left by the user?

Reply:
Yes, it will handle a single stray `,`